example/src/main/java/io/netty/example/stomp/websocket/StompWebSocketClientPageHandler.java

```diff
@@ -0,0 +1,132 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.example.stomp.websocket;
+
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.DefaultFileRegion;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.DefaultHttpResponse;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpResponse;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.LastHttpContent;
+import io.netty.util.CharsetUtil;
+
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.net.URL;
+
+import static io.netty.handler.codec.http.HttpHeaderNames.*;
+import static io.netty.handler.codec.http.HttpHeaderValues.KEEP_ALIVE;
+import static io.netty.handler.codec.http.HttpHeaderValues.*;
+import static io.netty.handler.codec.http.HttpResponseStatus.*;
+import static io.netty.handler.codec.http.HttpVersion.*;
+
+@Sharable
+public final class StompWebSocketClientPageHandler extends SimpleChannelInboundHandler<FullHttpRequest> {
+
+    static final StompWebSocketClientPageHandler INSTANCE = new StompWebSocketClientPageHandler();
+
+    private StompWebSocketClientPageHandler() {
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest request) {
+        if (request.headers().contains(HttpHeaderNames.UPGRADE, HttpHeaderValues.WEBSOCKET, true)) {
+            ctx.fireChannelRead(request.retain());
+            return;
+        }
+
+        if (!sendResource(request, ctx)) {
+            FullHttpResponse notFound = new DefaultFullHttpResponse(request.protocolVersion(), NOT_FOUND);
+            notFound.headers().set(CONTENT_TYPE, TEXT_PLAIN);
+            String payload = "Requested resource " + request.uri() + " not found";
+            notFound.content().writeCharSequence(payload, CharsetUtil.UTF_8);
+            HttpUtil.setContentLength(notFound, notFound.content().readableBytes());
+            sendResponse(notFound, ctx, true);
+        }
+    }
+
+    private static boolean sendResource(FullHttpRequest request, ChannelHandlerContext ctx) {
+        String requestResource = request.uri().substring(1);
```

Comment:
I think we need to check that `!....uri().isEmpty()` as otherwise it may produce an `IndexOutOfBoundsException` above.

Reply:
Done